File: c/gpio-led.c
#include <stdio.h>
#include <wiringPi.h>

#define LED 4

int main() {
    if (wiringPiSetup() == -1) return 1;

    pinMode(LED, OUTPUT);

    for (;;) {
        digitalWrite(LED, HIGH);
        printf("on\n");
        delay(1000);
        digitalWrite(LED, LOW);
        printf("off\n");
        delay(1000);
    }

    return 0;
}

File: javascript/gpio-led.js
#!/usr/local/bin/node
var LED = 4;

var WPI = require('wiring-pi');

// Init wiring-pi
if (WPI.wiringPiSetup() == -1) {
    console.error('wiring-pi setup failure');
    return;
}

WPI.pinMode(LED, WPI.OUTPUT);

for (;;) {
    WPI.digitalWrite(LED, WPI.HIGH);
    console.log('on');
    WPI.delay(1000)
    WPI.digitalWrite(LED, WPI.LOW);
    console.log('off');
    WPI.delay(1000)
}
